@xen-orchestra/web-core/lib/components/table/ColumnHeader.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+  <MenuList :disabled="disabled" placement="bottom-start" shadow>
+    <template #trigger="{ open, isOpen }">
+      <th
+        :class="{ interactive, disabled, focus: isOpen }"
+        class="column-header"
+        @click="ev => (interactive ? open(ev) : noop())"
+      >
+        <div class="content">
+          <span class="label">
+            <UiIcon :icon />
+            <slot />
+          </span>
+          <UiIcon v-if="currentInteraction !== undefined" :icon="currentInteraction.icon" />
```

Comment:
There is already a check inside the `UiIcon` component that will not render the component if the `icon` prop is `undefined`, so you can safely remove this one, just don't forget to add `?` to `currentInteraction`:
```suggestion
          <UiIcon :icon="currentInteraction?.icon" />
```